File: Blind 75/Array/53. Maximum Subarray.ts
function maxSubArray(nums: number[]): number {
  let subarraySum = [...nums];

  for (let i = 1; i < subarraySum.length; i++) {
    subarraySum[i] = Math.max(subarraySum[i], subarraySum[i] + subarraySum[i - 1]);
  }
  return Math.max(...subarraySum)
}

File: Blind Golang/Array/53. Maximum Subarray.go
package array

func maxSubArray(nums []int) int {
	maxSums := nums
	for i := 1; i < len(nums); i++ {
		if maxSums[i-1] > 0 {
			maxSums[i] += maxSums[i-1]
		}
	}
	return maxOfSlice(maxSums)
}

func maxOfSlice(nums []int) int {
	max := nums[0]

	for i := 1; i < len(nums); i++ {
		if cur := nums[i]; cur > max {
			max = cur
		}
	}

	return max
}
